### PR TITLE
Add checks for required parameters to Provider scripts (S-Z)

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #1883 by @obulat

## Description

This PR was originally created to replace the None checks for required values with `if <value>` checks because falsy values such as empty string or empty list should also short-circuit the provider API scripts.

After @AetherUnbound's suggestion to add tests, I split the PR into 3 to make the changes easier to review. This PR contains changes for the last 5 providers (from "S").

I wasn't aware about the recent addition of an Exception when one of the required param is missing. After this, we have to make sure that the provider API scripts check for truthiness of the required parameter values before returning the data in `get_record_data` function.

## Testing Instructions

The tests should pass, and the additional tests should test the changes.
Check that all of the possible "falsy" values are tested in the provider tests.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: catalog/dags/common/licenses/licenses.py [hunk 1]
```diff
@@ -25,6 +25,12 @@ class InvalidLicenseURLException(Exception):
 
 LicenseInfo = namedtuple("LicenseInfo", ["license", "version", "url", "raw_url"])
 
+# SMK sometimes uses incorrect URL for licenses
```
Comment (obulat) [inline catalog/dags/common/licenses/licenses.py:28]: Added in here because the example SMK used in test uses this ("share-your-work") URL, and the tests fail  because of this. Also added in the `LicenseURL` PR because that's where this change actually belongs. 

I do think that we should treat the works with this URL as marked with CC0 or PDM. If others think we shouldn't, I'll update the test object to use a valid CC license URL.

File: catalog/dags/providers/provider_api_scripts/smk.py [hunk 2]
```diff
@@ -87,41 +86,39 @@ def _get_creator(item: dict) -> str | None:
...
+
+        thumbnail_url = item.get("image_thumbnail")
+        height = item.get("image_height")
+        width = item.get("image_width")
+        filesize = item.get("image_size") or item.get("size")
 
         # We used to get additional images from the `alternative_images` field,
         # but we found these to be either redundant, duplicates, or lower quality
         # versions. See https://github.com/WordPress/openverse-catalog/issues/875
         # for the full investigation & discussion
```
Comment (stacimc) [inline catalog/dags/providers/provider_api_scripts/smk.py:110]: The method signature here confused me at first, but this comment explains it; this used to build a list including related images. Now that we only ever return one image, I think we should get rid of this method and pull this logic into `get_record_data`.

This isn't related to this refactor so we can do that in another PR, but it stood out to me so worth mentioning!

File: catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py [hunk 3]
```diff
@@ -107,6 +107,56 @@ def test_get_query_param_offset_page_number():
...
+@pytest.mark.parametrize(
+    "record",
+    [
+        # missing foreign_landing_url
+        {"links": {}},
+        {"links": {"self": ""}},
+        {"links": {"self": "link"}, "attributes": {"multimedia": []}},
+    ],
+)
+def test_get_record_data_returns_none_for_falsy_fid_and_landing_url(record):
```
Comment (stacimc) [inline catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py:119]: I didn't immediately understand why these cases are pulled out from the test case below. It's because when the foreign landing url or id are missing, we return `None` right away, but otherwise we attempt to build and return an image list -- which will be empty if the records were all invalid (missing image_url, etc). Could you add a comment to that effect? Or alternatively, I suppose they could all be in one test with `assert not actual_record_data`.

Comment (stacimc) [inline catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py:119]:
```suggestion
def test_get_record_data_returns_none_for_falsy_landing_url_and_multimedia(record):
```

Falsy fid is not tested in this case, right?

File: catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py [hunk 4]
```diff
@@ -107,6 +107,56 @@ def test_get_query_param_offset_page_number():
...
+        # missing foreign_identifier
+        {"links": {"self": "link"}, "attributes": {"multimedia": [{"admin": {}}]}},
+        {
+            "links": {"self": "link"},
+            "attributes": {"multimedia": [{"admin": {"uid": ""}}]},
+        },
+        {
+            "links": {"self": "link"},
+            "attributes": {"multimedia": [{"admin": {"uid": "fid"}}]},
+        },
```
Comment (stacimc) [inline catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py:136]: This test case is an example of missing image_url, right, not missing foreign_identifier?

File: catalog/tests/dags/providers/provider_api_scripts/test_smk.py [hunk 5]
```diff
@@ -132,3 +141,27 @@ def test_get_record_data_returns_main_image():
...
+@pytest.mark.parametrize(
+    "property_name",
+    [
+        pytest.param("object_number", id="falsy foreign_landing_url"),
+        pytest.param("rights", id="falsy license_info"),
+    ],
+)
+def test_get_record_data_returns_none_with_falsy_required_property(property_name):
+    item = _get_resource_json("item.json")
+    if property_name == "id":
```
Comment (stacimc) [inline catalog/tests/dags/providers/provider_api_scripts/test_smk.py:155]: What is this check doing?

File: catalog/tests/dags/providers/provider_api_scripts/test_wordpress.py [hunk 6]
```diff
@@ -61,11 +61,11 @@ def test_get_should_continue_checks_total_pages(
 
 
 @pytest.mark.parametrize("missing_field", ["slug", "link"])
-def test_get_record_data_returns_none_when_missing_necessary_data(
+def test_get_record_data_returns_none_when_necessary_data_is_falsy(
     ingester, missing_field
 ):
     image_data = _get_resource_json("full_item.json")
-    image_data.pop(missing_field, None)
+    image_data[missing_field] = ""
```
Comment (stacimc) [inline catalog/tests/dags/providers/provider_api_scripts/test_wordpress.py:68]: Could we add a check for when the field is missing, in addition to when it is falsy?

**Review by stacimc — COMMENTED** [review]
Thanks so much for all of this, @obulat :) I appreciate the split into smaller PRs as well, it has been very helpful for reviewing! I left some comments on test coverage but no real blocking concerns other than adding an `image_url` check for SMK.

File: catalog/dags/providers/provider_api_scripts/smk.py [hunk 7]
```diff
@@ -87,41 +86,39 @@ def _get_creator(item: dict) -> str | None:
...
-            )
+
+        if not (foreign_identifier := iiif_id or item.get("id")):
+            return []
+
+        if iiif_id:
+            image_url = SmkDataIngester._get_image_url(iiif_id)
+        else:
+            # Legacy images do not have IIIF links.
+            image_url = item.get("image_native")
```
[the remark this answers is not in the document]
Comment (obulat) [reply catalog/dags/providers/provider_api_scripts/smk.py:100]: Definitely! Thank you for catching :)

File: catalog/tests/dags/providers/provider_api_scripts/test_smk.py (hunk 5 above)
Comment (obulat) [reply catalog/tests/dags/providers/provider_api_scripts/test_smk.py]: It's left out from copy-pasting. I removed it, and improved tests

**Review by stacimc — APPROVED** [review]
Looks great! Thanks for cleaning up SMK as well :)

File: catalog/tests/dags/providers/provider_api_scripts/test_wikimedia_commons.py [hunk 8]
```diff
@@ -249,6 +249,33 @@ def test_extract_title_gets_cleaned_title(wmc):
     assert actual_title == expected_title
 
 
+def test_get_record_data_returns_none_when_missing_foreign_identifier(wmc):
+    media_data = _get_resource_json("image_data_example.json")
+    media_data["pageid"] = ""
+
+    record_data = wmc.get_record_data(media_data)
+
+    assert record_data is None
```
Comment (krysal) [inline catalog/tests/dags/providers/provider_api_scripts/test_wikimedia_commons.py:258]: Can't this be included as a parameter in the test below?

**Review by krysal — APPROVED** [review]
LGTM! I left a few comments but not consider those blocking. Great job with the None <> Falsy check changes for all the DAGs! 👏

File: catalog/tests/dags/providers/provider_api_scripts/test_wikimedia_commons.py (hunk 8 above)
Comment (obulat) [reply catalog/tests/dags/providers/provider_api_scripts/test_wikimedia_commons.py:258]: It's separate because the level of nesting for the parameters is different: `pageid` is the top-level property, and others are nested in the first item of `imageinfo`. I could combine them, but that would complicate the test functions.

File: catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py [hunk 9]
```diff
@@ -107,6 +107,63 @@ def test_get_query_param_offset_page_number():
...
+        {"links": {}},
+        {"links": {"self": ""}},
+        {"links": {"self": "link"}, "attributes": {"multimedia": []}},
+    ],
+)
+def test_get_record_data_returns_none_for_falsy_foreign_landing_url_and_multimedia(
+    record,
+):
+    actual_record_data = sm.get_record_data(record)
+    assert actual_record_data is None
```
[the remark this answers is not in the document]
Comment (obulat) [reply catalog/tests/dags/providers/provider_api_scripts/test_science_museum.py:126]: It's more difficult to parametrize these tests because of the different nesting levels: we would have to somehow get the correct nested property and set it to null. I've used the order in which we extract the properties here, so that `foreign_landing_url` is the first thing that is checked in the provider script, and the test will return `None` right away. For other tests, `foreign_landing_url` is always available because they first check for it before checking other parameters.

We could create a separate issue to improve the tests here, what do you think?
